lightning/src/ln/channel.rs

```diff
@@ -3146,7 +3145,9 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		}
 		if header.bitcoin_hash() != self.last_block_connected {
 			self.last_block_connected = header.bitcoin_hash();
-			self.channel_monitor.last_block_hash = self.last_block_connected;
+			if let Some(channel_monitor) = self.channel_monitor.as_mut() {
+				channel_monitor.last_block_hash = self.last_block_connected;
```

Comment:
710520b

I found this a bit confusing given that you already feed ChannelMonitor with block in any ManyChannelMonitor implementation (and our API should allow block skew for watchtowers having a different chain access than client)

Reply:
This should just be dropped when we remove Channel's local ChannelMonitor copy, its just a historical quirk (and note that the internal copy is never used for anything other than local transaction creation now).